<h1>Case Study Scenario</h1>

· Client “ABC” is currently investing 500k SGD in LinkedIn Marketing Solutions. It’s renewal time and the client wishes to decrease their investment by half as they don’t see the value of the product.  

· The account manager for this account has asked you to provide a return on investment analysis to help showcase the value the client is getting from their investment.  

· Based on the data given to you in the excel document attached and any other data points you can pull from our network, please create a Python Jupyter Notebook or similar solution to produce a series of data analysis, visualizations, or other outputs. The output of this notebook should help demonstrate the value of the client’s investment and any recommendations you would have to help prevent this churn.  

· Select any client of your choice and make whatever assumptions you feel is necessary. You may also include any additional research you feel would help support your analysis to prove the value of this investment.  

· Remember to think creatively with the data and analysis you develop and think strategically about the recommendation you are proposing  

· Please ensure you also list any questions you would ask the Account Manager to help you scope out this request.   

· Finally, please list any recommendations you would have for the account manager moving forward to help prevent churn, before being challenged like this by the client.  

<h2>Objective</h2>
To visualize the shape of the data for report analysis

<i>Details to be obtained for in-depth investigation from Account Manager:</i>
- CompanyABC's business nature: product / services? (Content Targeting / Regions / Localisation)
- Purpose(s) of the campaign setup (Promotion / Brand Awareness / Company Growth / Website Traffic)
- How is CompanyABC's campaign setup (Audience Targeting / Qualifier)
- Focuses of the campaign (Impression / Clicks / Conversions)
- Duration of the campaign (Frequency & Redundancy)
- Bid/Spent per Campaign (CPM, CPC)

In [1]:
import pandas as pd

total_investment = 500000
# Load Data
df = pd.read_excel('Sponsored Updates - LMS.xlsx', index_col=None, header=0, dtype={'CAMPAIGN_ID': str})
# del df['COMPANY']
# del df['CAMPAIGN_ID']
df['CTR'] = round((df['CLICKS'] / df['IMPRESSIONS']) * 100, 2)
df['MONTH'] = df['DAY'].dt.month
df['YEAR'] = df['DAY'].dt.year
df['DAY_OF_MONTH'] = df['DAY'].dt.day
df['DAY_OF_WEEK'] = df['DAY'].dt.day_name()

df = df.astype({'DAY_OF_MONTH': 'str', 'YEAR': 'str', 'MONTH': 'str', 'CAMPAIGN_ID': 'str'})
df.head()

COMPANY        DAY CAMPAIGN_ID CONTENT_SHARED  IMPRESSIONS  CLICKS  \
0  Company_ABC 2017-03-31        3551        Picture        52190    1396   
1  Company_ABC 2017-03-31        2344          Video        15765    1234   
2  Company_ABC 2017-03-30        2918          Video        78998    1204   
3  Company_ABC 2017-03-30        4503        Article        19506    1492   
4  Company_ABC 2017-03-29        2735        Picture        71371     811   

   LIKES  COMMENTS  SHARES  RESHARES   CTR MONTH  YEAR DAY_OF_MONTH  \
0    138       104      83        14  2.67     3  2017           31   
1    149        94      76         2  7.83     3  2017           31   
2    122       167     119        30  1.52     3  2017           30   
3    196       108      27        40  7.65     3  2017           30   
4    102        67      18         3  1.14     3  2017           29   

  DAY_OF_WEEK  
0      Friday  
1      Friday  
2    Thursday  
3    Thursday  
4   Wednesday

In [2]:
df.groupby(['YEAR', 'MONTH', 'DAY_OF_MONTH', 'CONTENT_SHARED']).sum().filter(regex='^(?!CTR).*$', axis='columns')

IMPRESSIONS  CLICKS  LIKES  COMMENTS  \
YEAR MONTH DAY_OF_MONTH CONTENT_SHARED                                         
2016 1     1            Article              143839    3383    515       406   
           10           Picture              120733    1615    264       281   
                        Video                 14981     455    136       187   
           11           Article               32164    1224    131       135   
                        Picture               69308    1908    305       351   
...                                             ...     ...    ...       ...   
2017 3     8            Article               24481    1440    193       107   
                        Picture               13940     740    144        85   
                        Video                 75780     863    158       136   
           9            Article              103686    1470    280       278   
                        Picture               29119     671    196        65   

                                        SHARES  RESHARES  
YEAR MONTH DAY_OF_MONTH CONTENT_SHARED                    
2016 1     1            Article            275       115  
           10           Picture            214        58  
                        Video              162        23  
           11           Article             75        44  
                        Picture             90        35  
...                                        ...       ...  
2017 3     8            Article            163        24  
                        Picture            103         2  
                        Video               87        50  
           9            Article            280        79  
                        Picture             61        27  

[974 rows x 6 columns]

Assuming ads conversions are Likes, Comments, Shares, Reshares

In [3]:
df.groupby(['CONTENT_SHARED']).sum().filter(regex='^(?!CTR).*$', axis='columns')

IMPRESSIONS  CLICKS  LIKES  COMMENTS  SHARES  RESHARES
CONTENT_SHARED                                                        
Article            22923558  479534  76241     62591   53213     12216
Picture            23962154  482333  77144     64799   52004     12740
Video              21882500  465449  72370     60063   50663     11646

In [4]:
df.groupby(['COMPANY']).sum().filter(regex='^(?!CTR).*$', axis='columns')

IMPRESSIONS   CLICKS   LIKES  COMMENTS  SHARES  RESHARES
COMPANY                                                              
Company_ABC     68768212  1427316  225755    187453  155880     36602

In [5]:
df.groupby(['COMPANY']).sum().filter(regex='^(?!CTR).*$', axis='columns').apply(lambda f: total_investment/f)

IMPRESSIONS    CLICKS    LIKES  COMMENTS    SHARES   RESHARES
COMPANY                                                                   
Company_ABC     0.007271  0.350308  2.21479  2.667335  3.207596  13.660456

<h2>Value of Investment</h2>

- Cost per Impression
    500k / Total Impressions = SGD 0.007

- Cost per Click
    500k / Total Clicks = SGD 0.35

- Cost per Like
    500k / Total Likes = SGD 2.2
    
- Cost per Comments
    500k / Total Comments = SGD 2.667
    
- Cost per Shares
    500k / Total Shares = SGD 3.208
    

In [6]:
from IPython.display import display, Markdown, Latex
total_clicks = df['CLICKS'].sum()
total_impressions = df['IMPRESSIONS'].sum()
avg_ctr = round((total_clicks / total_impressions) * 100, 2)

display(Markdown('Total Clicks: {0}'.format(total_clicks)))
display(Markdown('Total Impressions: {0}'.format(total_impressions)))
display(Markdown('Average CTR: {0} %'.format(avg_ctr)))


Total Clicks: 1427316

Total Impressions: 68768212

Average CTR: 2.08 %

In [ ]:
from jupyter_dash import JupyterDash
import dash_core_components as dcc
import dash_html_components as html
from dash.dependencies import Input, Output
import plotly.express as px

# Build App
app = JupyterDash(__name__)

colors = {
    'background': '#FFFFAA',
    'text': '#f0501a'
}

app.layout = html.Div(style={'backgroundColor': colors['background']}, children=[
    html.H1(
        children='COMPANY ABC',
        style={
            'textAlign': 'center',
            'color': colors['text']
        }
    ),
     html.Div(children='Investment Dashboard (Area)', style={
        'textAlign': 'center',
        'color': colors['text']
    }),
    dcc.Dropdown(
        id="dropdown",
        options=[{"label": x, "value": x} 
                 for x in df.columns[4:11]],
        value=df.columns[1],
        clearable=False,
    ),
    dcc.Graph(id="time-series-chart"),
])

# Run app and display result inline in the notebook
@app.callback(
    Output("time-series-chart", "figure"), 
    [Input("dropdown", "value")])

def display_time_series(dropdown):
    fig = px.area(df, facet_col="CONTENT_SHARED", facet_col_wrap=1, x='DAY', y=dropdown)
    return fig


app.run_server(mode='inline')

In [ ]:
# Build App
app = JupyterDash(__name__)

colors = {
    'background': '#FFFFAA',
    'text': '#f0501a'
}

content_shared = df.CONTENT_SHARED.unique()

app.layout = html.Div(style={'backgroundColor': colors['background']}, children=[
    html.H1(
        children='COMPANY ABC',
        style={
            'textAlign': 'center',
            'color': colors['text']
        }
    ),
     html.Div(children='Investment Dashboard (Line)', style={
        'textAlign': 'center',
        'color': colors['text']
    }),
    dcc.Dropdown(
        id="dropdown",
        options=[{"label": x, "value": x} 
                 for x in df.columns[4:11]],
        value=df.columns[1],
        clearable=False,
    ),
    dcc.Checklist(
        id="checker",
        options=[{"label": x, "value": x} 
                 for x in content_shared],
        value=content_shared,
        labelStyle={'display': 'inline-block'}
    ),
    dcc.Graph(id="time-series-chart"),
])

# Run app and display result inline in the notebook
@app.callback(
    Output("time-series-chart", "figure"), 
    [Input("dropdown", "value"), Input("checker", "value")])

def display_time_series(dropdown, checker):
    fig = px.line(df[df['CONTENT_SHARED'].isin(checker)], x='DAY', y=dropdown)
    return fig


app.run_server(mode='inline')

<h2>Observation (Bottleneck)</h2>
<p>
Noticed that throughout the campaigns period, we didn't see much growth on the marketing metrics.
We can safely conclude that the campaign setup needs be fine tuned for the effectiveness of campaign's audience targeting.
</p>    

<i>As mentioned earlier on, some details need to be obtained for in-depth investigation:</i>
- CompanyABC's business nature: product / services? (Content Targeting / Regions / Localisation)
- Purpose(s) of the campaign setup (Promotion / Brand Awareness / Company Growth / Website Traffic)
- How is CompanyABC's campaign setup (Audience Targeting / Qualifier)
- Focuses of the campaign (Impression / Clicks / Conversions)
- Duration of the campaign (Frequency & Redundancy)
- Bid/Spent per Campaign (CPM, CPC)

<h2>Recommendation</h2>
<img src="https://content.linkedin.com/content/dam/me/business/en-us/marketing-solutions/a/ops/creative-submission-templates/campaigns-tab-cropped.png.original.png" alt="Campaign Manager"/>

<h2>Reporting and Analytics - Measure the ROI of your LinkedIn ads</h2>
1. Measure conversions
<ul>Use <a href="https://business.linkedin.com/marketing-solutions/conversion-tracking">conversion tracking</a> to understand how your ads are driving business results, like generating leads, purchases, and event registrations.</ul>
2. Analyze performance
<ul>Compare metrics, like clicks, impressions, and social actions, to learn which campaigns and ad creatives are most effective.</ul>
3. Understand your audience
<ul>Learn more about your audience, on and off LinkedIn:<br/>
Use <a href="https://www.linkedin.com/help/linkedin/answer/61000/demographics-for-your-ads?src=or-search&veh=www.google.com%7Cor-search&trk=nw_ml_at">campaign demographics</a> to see who engages with your ads. View by traits like job titles, company names, and industries.<br/>
Discover the professional traits and content preferences of your website visitors with <a href="https://business.linkedin.com/marketing-solutions/website-demographics">Website Demographics</a>.
</ul>
4. Keep a pulse on your biggest opportunities
<ul>Use the <a href="https://business.linkedin.com/marketing-solutions/success/best-practices/track-and-measure-engagement">Company Engagement Report to measure engagement with your brand across LinkedIn and your website. See website visits, ad engagement and organic post engagement by target company.</ul>

<h2>Analyze your campaign performance - Audience Targeting</h2>

1. Choose your objective
2. Identify your key metrics based on the campaign goal
3. Set up conversion tracking and Lead Gen Forms
4. Evaluate ad performance
5. Optimize with demographic insights
6. Optimize with performance insights

<h3>References</h3>

<a href="https://business.linkedin.com/marketing-solutions/reporting-analytics">https://business.linkedin.com/marketing-solutions/reporting-analytics</a><br/>
<a href="https://business.linkedin.com/marketing-solutions/success/best-practices/analyze-your-performance">https://business.linkedin.com/marketing-solutions/success/best-practices/analyze-your-performance</a><br/>
<a href="https://business.linkedin.com/marketing-solutions/success/best-practices/conversion-tracking-tips">https://business.linkedin.com/marketing-solutions/success/best-practices/conversion-tracking-tips</a><br/>
<a href="https://business.linkedin.com/marketing-solutions/success/best-practices/ad-targeting-best-practices">https://business.linkedin.com/marketing-solutions/success/best-practices/ad-targeting-best-practices</a><br/>
<a href="https://business.linkedin.com/marketing-solutions/success/best-practices/track-and-measure-engagement">https://business.linkedin.com/marketing-solutions/success/best-practices/track-and-measure-engagement</a><br/>